src/ui/screens/AddAccountScreen/AddAccountScreen.tsx

```diff
@@ -59,10 +59,14 @@ export function AddAccountScreen({navigation}: {navigation: NavigationProp<AppSt
 
   const handleScan = useCallback(
     ({data}) => {
-      const parsed = parseAddress(data);
-      if (isAddressValid(currentNetwork, parsed.address)) {
-        dispatch({type: 'SET_ADDRESS', payload: parsed.address});
-        dispatch({type: 'SET_STEP', payload: 'preview'});
+      try {
+        const parsed = parseAddress(data);
+        if (isAddressValid(currentNetwork, parsed.address)) {
```

Comment:
When this is false, we should also display an error message as well, explaining that the address scanned is for another network